### PR TITLE
Ruby Resorts Back End

### PR DESCRIPTION
**Pull request by anthonymr**
Hi, we are submitting the first fully working version to the Ruby Resorts Project in this PR.

[Live demo](https://ruby-resorts-front-end.onrender.com/)
[Front end pull request](https://github.com/anthonymr/ruby-resorts-frontend/pull/34)

### ⭐  Test admin user
  - username: admin
  - password: admin123

### 👥  Team members

We are a team of four: @anthonymr @gsmalik030 @RaphDasilva @ram1117 . 

### ➕  Aditional requirements

We have implemented the following as additional requirements in our project.
  1. Implement proper user authentication from the front end to the server.
  2. Make sure that the "Add item" and "Delete item" links are accessible only by users who are admins.

### ✅ Back end implementation

We have implemented the following in the back end:
  1. Fully user authentication system using `JWT`.
  2. Ability to create a new user.
  3. Ability to get Hotels (with cities) list.
  4. Ability to get, create, and delete Rooms. Create and delete only for admins.
  5. Ability to get all, get one, create and delete Reservations.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,233 @@
...
+
+### Run tests
+
+To run tests, run the following command:
+
+```sh
+  rspec spec
+```
+
+we have added Rspec test cases for all models and API endpoints in the project.
```
Comment (Meltrust) [inline README.md:179]: - [x]  You are using Rspec, but the "test" folder still exists, which is used by another test framework. Kindly consider deleting that folder if you are not using it. 👍

File: README.md [hunk 2]
```diff
@@ -1,2 +1,233 @@
...
+```
+
+The above command installs necessary gems used in the project
+
+### Usage
+
+To run the project, execute the following command:
+
+```sh
+  npm rails s
```
Comment (Meltrust) [inline README.md]: - [x] This command does not seem to be working.  The correct command would be just `rails s`.  Kindly correct that command on your readme. 👍

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @anthonymr,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Tests are passing✔️
-  Nice code organization ✔️ 
-  App is working well✔️ 
-  Good readme ✔️ 




#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

File: README.md (hunk 1 above)
Comment (anthonymr) [reply README.md:179]: Hi @Meltrust, thanks very mucho for the advice, we deleted the folder and everything seems working well.

File: README.md (hunk 2 above)
Comment (anthonymr) [reply README.md]: Hi @Meltrust, you were right. We changed that command and also added extra guidance about first-time DB creation.

**Review by Meltrust — APPROVED** [review]
#### Hi @anthonymr,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight

- Tests are passing✔️ 
- Nice code implementation✔️ 
- Nice docs✔️ 
- Good readme and instructions ✔️ 
- Readme has been corrected ✔️ 
- Superfluous `test` folder has been removed ✔️ 
- Good job ✔️ 



Cheers and Happy coding!👏👏👏


------

**Pull request merged**